api/v1beta2/cryostat_types.go

```diff
@@ -587,6 +591,12 @@ type CertificateSecret struct {
 	CertificateKey *string `json:"certificateKey,omitempty"`
 }
 
+type DeclarativeCredential struct {
+	// Name of secret in the local namespace
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
+	SecretName string `json:"secretName"`
```

Comment:
It might be a good idea to include details of the expected format of the secret in the comment. These comments get translated into UI labels.